RA Scripts/Legend of Zelda, The The Minish Cap.rascript

```diff
@@ -761,7 +769,8 @@ function remoteBombs() => bit0(0x00ab34)
 achievement(
     title = "Don't Hurt Yourself", description = "Obtain a Bomb Bag.", points = 5,
     id = 2056, badge = "155812", published = "8/4/2013 8:02:32 AM", modified = "1/30/2021 2:30:23 PM",
-    trigger = WasValueJustSetInGame(bombs(), 0, 1) && Delta(remoteBombs()) == 0
+    trigger = WasValueJustSetInGame(bombs(), 0, 1) && Delta(remoteBombs()) == 0 &&
+              MusicId() == 0x14 && ScreenIdWord() == 0x2620
```

Comment:
This code does not, by my view, actually add network error protection, as all it does is add additional AND'd conditions instead of adding alts that could also trigger it. In addition, if an achievement is triggered on the client, it will be considered in a Triggered state and will not actually try triggering again when the alternative trigger method is also triggered, unless the player has Keep Active checked.

I do think additional protection is a worthy endeavor, but I don't think this is the correct way to go about it.

Reply:
As I stated in the ticket, I felt the code was fine, and I couldn't reproduce it. Still, I prefer to tighten logic when I can. If this change was unnecessary, I'm fine removing it. The ticket is still considered closed as unable to reproduce, which is true.

Reply:
Tightening logic can actually be risky, as it potentially leads to a situation where an achievement won't trigger at all. Now, I don't think this will happen here, as I'm sure you tested the code, but considering what this change actually does, I don't think it adds the benefit that was mentioned in the rationale. If you can't reproduce a problem with code that otherwise looks fine, it should never be changed (if you're optimizing, that's different; then the code isn't "fine," you could argue).

I'm not saying it's bad _per se_ to add network error protection; what I am saying is that this code change doesn't actually do that and should either be changed to do that (see my reasoning above as to why our success to that effect may be limited here) or removed outright.